### PR TITLE
Visual updates to `Button` & `IconButton` components

### PR DESCRIPTION
**Pull request by robphoenix**
- [x] Font weight - change from Semibold to Medium (same across all)
- [x] Font size - change to 16px (same across all sizes)
- [x] Icons - change from 24px to 16px across all sizes
- [x] Spacing between text and icon (gap) - change from 16px to 8px
- [x] Align icon colour with text colour
- [x] Change min width to 136px for large and medium and small to 120px
- [x] Add inverted prop
- [x] Updated focus visible style (focus outline offset is 2px)
- [x] Align colours with App UI where possible
- [x] Defined transition 👉  [UWDS-2575](https://linear.app/utilitywarehouse/issue/UWDS-2575/add-transition-to-buttons-for-hover-and-active-interactions)

### CODE REVIEW

File: packages/web-ui/src/Button/Button.docs.mdx [hunk 1]
```diff
@@ -138,18 +152,15 @@ can be used to display different sizes at different breakpoints.
 You can nest icons directly inside the `Button`. An appropriate gap is provided
 automatically, and the icon will inherit the appropriate colours.
 
-This button is intended to be used with the [UW Icons](https://uw-icons.vercel.app/) packages. Small buttons
-should use 16px icons (small), while medium buttons should use 24px icons
-(medium).
-
-Using other icons may result in an unexpected appearance.
+This button is intended to be used with the [UW Icons](https://uw-icons.vercel.app/) packages.
+All links should use the small (16px) icons.
```
Comment (noeliacabane) [inline packages/web-ui/src/Button/Button.docs.mdx:156]:
```suggestion
All buttons should use the small (16px) icons.
```

File: packages/web-ui/src/Button/Button.docs.mdx [hunk 2]
```diff
@@ -138,18 +152,15 @@ can be used to display different sizes at different breakpoints.
 You can nest icons directly inside the `Button`. An appropriate gap is provided
```
Comment (noeliacabane) [inline packages/web-ui/src/Button/Button.docs.mdx:152]: Could we add examples with icons on the left as well? And let's omit the word 'page' in the examples, as it suggests a specific use case. We can use the words 'next' and 'back' instead

Screenshot 2024-09-26 at 08 59 43

File: packages/web-ui/src/BaseButton/BaseButton.tsx [hunk 3]
```diff
@@ -34,106 +36,204 @@ const classSelectors = {
 
 const StyledElement = styled(UnstyledButton)<BaseButtonProps>(() => {
   return {
+    transitionDuration: '0.3s',
+    transitionProperty: 'color, background-color, border-color',
```
Comment (noeliacabane) [inline packages/web-ui/src/BaseButton/BaseButton.tsx:40]: 💅 beautiful

**Comment by noeliacabane** [comment]
Testing in safari:
- when I use the code inspector to see the `focus-visible` state it shows the outline/ring
- when I use tab/keyboard to navigate the page it doesn't show the outline/ring 🤔 (Same happens in other components like Link) or scrolls down. 

Screenshot 2024-09-26 at 09 31 32

**Review by noeliacabane — APPROVED** [review]
All the comments can be addressed in this PR or in a follow up one 
Design QA done on: brave/chrome, safari and firefox

**Comment by noeliacabane** [comment]
Is code connect missing still for IconButton?

Screenshot 2024-09-26 at 09 59 52

**Comment by robphoenix** [comment]
> Is code connect missing still for IconButton?

yah, should be there
image

**Comment by robphoenix** [comment]
> Is code connect missing still for IconButton?

it was, I was using the wrong node id 🤦🏻 all fixed now @noeliacabane

**Pull request merged**